src/cached_fs_storage.rs

```diff
@@ -0,0 +1,402 @@
+/*
+ * Copyright 2018 Intel Corporation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * ------------------------------------------------------------------------------
+ */
+
+use std::collections::VecDeque;
+use std::io;
+use std::path::PathBuf;
+use std::sync::{Arc, RwLock, RwLockReadGuard, RwLockWriteGuard};
+
+use raft::{
+    self, eraftpb::{ConfState, Entry, HardState, Snapshot}, RaftState, Storage,
+};
+
+use fs_storage::FsStorage;
+use storage::StorageExt;
+
+pub const CACHE_SIZE: u64 = 10000;
+
+struct CacheStorage {
+    entries: VecDeque<Entry>,
+    // TODO: implement cache size limitation
```

Comment:
This cache doesn't seem to consume very much memory; I'd like some feedback on whether a max size should need to be enforced.

Reply:
Remove this TODO

Reply:
What are the items stored in this cache?  Blocks? Peers?  One may need enforcement, the other may not.

Reply:
This is essentially just storing block numbers. It has very low overhead. On a run with 50k+ blocks, the raft process as a whole only consumed about 35MB of memory with this cache.

Reply:
Ok.  Just remove the TODO, then.